libcudacxx/test/atomic_codegen/atomic_add_volatile.cu

```diff
@@ -0,0 +1,18 @@
+#include <cuda/atomic>
+
+__global__ void add_relaxed_device_volatile(int* data, int n) {
+    if (n) {
+        volatile auto ref = cuda::atomic_ref<int, cuda::thread_scope_device>{*(data)};
```

Comment:
From offline discussion, we shouldn't be supporting `volatile` atomic_ref objects because, unlike `atomic`, there are no `volatile` qualified member functions of `atomic_ref`. 

```suggestion
        auto ref = cuda::atomic_ref<int, cuda::thread_scope_device>{*(data)};
```

Reply:
Removed the volatile `atomic_ref` APIs. If anyone is using them, they'll encounter errors.